cmd/http.go

```diff
@@ -44,11 +44,19 @@ func loadTLSConfig(certPath, keyPath string) (*tls.Config, error) {
 	}, nil
 }
 
-func setupHTTPServer(ctx context.Context, cfg *apiServerConfig) (io.Closer, io.Closer, error) {
-	var (
-		podS     *http.Server
-		metricsS *http.Server
-	)
+func setupHTTPServer(ctx context.Context, cfg *apiServerConfig) (cancel func(), retErr error) {
+	var closers []io.Closer
+	cancel = func() {
+		for _, c := range closers {
+			c.Close()
+		}
+	}
+	defer func() {
+		if retErr != nil {
+			cancel()
```

Comment:
There's no more the need to define `cancel` func then. Can we just move the `for` loop above here?

Reply:
Sorry, we still need cancel because it is returned to the caller and used here.

Reply:
I get it now. If it errors, cancel happens immediately. If not, the caller will cancel later.